lib/error-helper.js

```diff
@@ -18,6 +23,26 @@ const checkErrorResponse = function(badgeData, err, res, notFoundMessage = 'not
   }
 };
 
+checkErrorResponse.asPromise = function ({ notFoundMessage }) {
+  return async function ({ buffer, res }) {
+    if (res.statusCode === 404) {
+      throw new NotFound(notFoundMessage);
+    } else if (res.statusCode !== 200) {
+      throw new InvalidResponse();
+    }
+    return { buffer, res };
+  };
+};
+
+async function asJson({ buffer, res }) {
+  try {
+    return JSON.parse(buffer);
+  } catch (err) {
+    throw InvalidResponse('');
```

Comment:
Should this be `throw new InvalidResponse()` here?

Reply:
Indeed!